extensions/guacamole-auth-radius/src/main/java/org/apache/guacamole/auth/radius/RadiusConnectionService.java

```diff
@@ -0,0 +1,352 @@
+import net.jradius.packet.AccessChallenge;
+import net.jradius.packet.RadiusResponse;
+
+/**
+ * Service for creating and managing connections to RADIUS servers.
+ */
+public class RadiusConnectionService {
+
+    /**
+     * Logger for this class.
+     */
+    private final Logger logger = LoggerFactory.getLogger(RadiusConnectionService.class);
+
+    /**
+     * Service for retrieving RADIUS server configuration information.
+     */
+    @Inject
+    private ConfigurationService confService;
+
+
+    /**
+     * The RADIUS client;
+     */
+    private RadiusClient radiusClient;
+
+    /**
+     * Creates a new instance of RadiusClient, configured with parameters
+     * from guacamole.properties.
+     *
+     * @throws GuacamoleException
+     *     If an error occurs while parsing guacamole.properties, or if the
+     *     configuration of RadiusClient fails.
+     */
+    private void createRadiusConnection() {
+
+        // Create the RADIUS client with the configuration parameters
+        try {
+            radiusClient = new RadiusClient(InetAddress.getByName(confService.getRadiusServer()),
+                                            confService.getRadiusSharedSecret(),
+                                            confService.getRadiusAuthPort(),
+                                            confService.getRadiusAcctPort(),
+                                            confService.getRadiusTimeout());
+        }
+        catch (GuacamoleException e) {
+            logger.error("Unable to initialize RADIUS client: {}", e.getMessage());
+            logger.debug("Failed to init RADIUS client.", e);
+        }
+        catch (UnknownHostException e) {
+            logger.error("Unable to resolve host: {}", e.getMessage());
+            logger.debug("Failed to resolve host.", e);
+        }
+        catch (IOException e) {
+            logger.error("Unable to communicate with host: {}", e.getMessage());
+            logger.debug("Failed to communicate with host.", e);
+        }
+
+    }
+
+    /**
+     * Creates a new instance of RadiusAuthentictor, configured with
+     * parameters specified within guacamole.properties.
+     *
+     * @return
+     *     A new RadiusAuthenticator instance which has been configured
+     *     with parameters from guacamole.properties, or null if
+     *     configuration fails.
+     */
+    private RadiusAuthenticator setupRadiusAuthenticator() throws GuacamoleException {
+
+        // If we don't have a radiusClient object, yet, don't go any further.
+        if (radiusClient == null) {
+            logger.error("RADIUS client hasn't been set up, yet.");
+            logger.debug("We can't run this method until the RADIUS client has been set up.");
+            return null;
+        }
+
+        // Pull configuration parameters from guacamole.properties
+        LocalEnvironment guacEnv = new LocalEnvironment();
+        String guacHome = guacEnv.getGuacamoleHome().getAbsolutePath();
```

Comment:
There's no need to convert the `File` returned by `getGuacamoleHome()` into a path string. There is a `File` constructor which accepts another `File` as the parent:

https://docs.oracle.com/javase/7/docs/api/java/io/File.html#File(java.io.File,%20java.lang.String)